### ISSUE TITLE
fix(jans-tent): no scope when using dynamic registration

### ISSUE THREAD

**Issue by manojs1978** [JanssenProject/jans#4615]
**Describe the bug**
getting  "error": "403 Client Error in jans-tent flow

**To Reproduce**

Go to jans tent
configure according to steps mentioned
https://github.com/JanssenProject/jans/blob/main/demos/jans-tent/README.md#jans-tent
modify config.py for ACR for simple_password_auth 
create client
run main.py
access https://localhost:9090/

**Expected behavior**
should able to complete jans-tent acr=simple_password flow

**Screenshots**
![image](https://user-images.githubusercontent.com/5729240/232446682-03ba97f9-dd64-45c7-8ab1-9c2af2eb2670.png)

**Desktop (please complete the following information):**
  - OS: suse
  - Version 15

**Smartphone (please complete the following information):**

**Additional context**

**Comment by manojs1978** [comment]
client created using register_new_client.py is not having scope due to which getting error.
manually added scope and i am able to complete flow.

![image](https://user-images.githubusercontent.com/5729240/232462841-632bccd0-ae60-4fa2-b29a-4ea26f4fe44f.png)

**Comment by christian-hawk** [comment]
I don't get it. Scope `openId` should be there by default, as it was should be there by default. 

Please provide the requested details on issue opening:

- What jans version are you using? It worked fine on 1.0.10.

So please notice, registration happens and `openid` scope is not there, is not a `jans-tent` (the reliable test tool) issue, but the services being tested with it. In that case, you need to transfer this issue to Jans-auth or something related.

**Comment by nynymike** [comment]
Chris, is your client requesting `openid` scope during dynamic client registration? It is allowed by default, so it shouldn't be an error on the AS side.

**Comment by manojs1978** [comment]
> I don't get it. Scope `openid` should be there by default, as it was should be there by default.
> 
> Please provide the requested details on issue opening:
> 
> * What jans version are you using? It worked fine on 1.0.10.
> 
> So please notice, registration happens and `openid` scope is not there, is not a `jans-tent` (the reliable test tool) issue, but the services being tested with it. In that case, you need to transfer this issue to Jans-auth or something related.
> 
> Anyway, checking test-client.log and jans logs, finding the related lines to the client registration and adding them here will also help jans team to help. Maybe an issue about default values for dynamic client registration, instead of only one as in #4296 ?

getting issue with jans 1.0.12 nightly build

**Comment by christian-hawk** [comment]
@nynymike @manojs1978 @moabu changed the issue tittle to the correct scope.

**When using dynamic registration endpoint, using tent, postman, curl or whatever, the `openid` scope is not in the registered client.**

To reproduce: 

1. try to register a client using dynamic registration endpoint
2. Open client using tui.
3. Client will not have TUI

Version tested: 1.0.12, ubuntu 20.0.4


### Additional context
Request params:
```json
{
	"redirect_uri": ["https://localhost:9090/oidc_callback"],
	"response_types": ["code"],
	"grant_types": ["authorization_code"],
	"application_type": "web",
	"client_name": "Jans Tent",
	"token_endpoint_auth_method": "client_secret_post"
}
```


@manojs1978

**Comment by yuriyz** [comment]
Exactly, `openid` scope is not in request thus Server will not grant it to client. AS behavior was changed in [#4426](https://github.com/JanssenProject/jans/issues/4426).

**Comment by nynymike** [comment]
Yuriyz is correct. Scope should be explicitly specified. There is a feature flag to maintain old behavior, but it should be disabled by default.

**Comment by christian-hawk** [comment]
@yuriyz @nynymike @manojs1978  It's probably my fault. 

What brought me to my misunderstanding:

- OpenId Dynamic Registration Specs don't mention scopes at all, in any of the examples.
- As some of the OP/AS and jans, till before #4426, accepted `openid` as default scope.

After researching OAuth docs, in [RFC7591 Section 1.3 ](https://datatracker.ietf.org/doc/html/rfc7591#section-1.3), I found the following:

> The following client metadata fields are defined by this
   specification.  The implementation and use of all client metadata
   fields is OPTIONAL, unless stated otherwise. 


> `scope`
> String containing a space-separated list of scope values (as
      described in [Section 3.3](https://datatracker.ietf.org/doc/html/rfc7591#section-3.3) of OAuth 2.0 [[RFC6749](https://datatracker.ietf.org/doc/html/rfc6749)]) that the client
      can use when requesting access tokens.  The semantics of values in
      this list are service specific.  If omitted, an authorization
      server MAY register a client with a default set of scopes.


So basically, it looks like:

1. `scope` is **not** required
2. It is **not mandatory** to the AS to define a default value, it MAY. 

I'm bringing it back to jans-tent context.

**Comment by christian-hawk** [comment]
Btw, I also found this issue that's interesting discussion about adding "non openid specs" params to DCR: 
https://bitbucket.org/openid/connect/issues/1060/update-dynamic-client-registration-spec-to
